cluster-operator/src/main/java/io/strimzi/operator/cluster/model/AbstractModel.java

```diff
@@ -408,21 +408,34 @@ public String parseLogging(Logging logging, ConfigMap externalCm) {
                 newSettings.addMapPairs(inlineLogging.getLoggers());
             }
 
-            return createPropertiesString(newSettings);
-
+            return createLog4jProperties(newSettings);
         } else if (logging instanceof ExternalLogging) {
             if (externalCm != null && externalCm.getData() != null && externalCm.getData().containsKey(getAncillaryConfigMapKeyLogConfig())) {
-                return externalCm.getData().get(getAncillaryConfigMapKeyLogConfig());
+                return maybeAddMonitorIntervalToExternalLogging(externalCm.getData().get(getAncillaryConfigMapKeyLogConfig()));
             } else {
                 log.warn("ConfigMap {} with external logging configuration does not exist or doesn't contain the configuration under the {} key. Default logging settings are used.",
                         ((ExternalLogging) getLogging()).getName(),
                         getAncillaryConfigMapKeyLogConfig());
-                return createPropertiesString(getDefaultLogConfig());
+                return createLog4jProperties(getDefaultLogConfig());
             }
 
         } else {
             log.debug("logging is not set, using default loggers");
-            return createPropertiesString(getDefaultLogConfig());
+            return createLog4jProperties(getDefaultLogConfig());
+        }
+    }
+
+    /**
+     * Adds 'monitorInterval=30' to external logging ConfigMap. If ConfigMap already has this value, it is persisted.
+     * @param data String with log4j2 properties in format key=value separated by new lines
+     * @return
+     */
+    protected String maybeAddMonitorIntervalToExternalLogging(String data) {
+        if (getAncillaryConfigMapKeyLogConfig().equals("log4j2.properties") && !data.contains("monitorInterval")) {
```

Comment:
I don't think this is really good enough. `monitorInterval` could appear in a log4j2 properties config in numerous places other than the top level. You should at least use a regex like `^monitorInterval\s+=` (though you should verify even that). Or we just parse it as an `OrderedProperties` and check for the `monitorInterval` key, which I see you're also doing further on....???